crates/router/src/connector/bluesnap/transformers.rs

```diff
@@ -0,0 +1,349 @@
+pub struct Card {
+    card_number: String,
+    expiration_month: String,
+    expiration_year: String,
+    security_code: String,
+}
+
+impl TryFrom<&types::PaymentsAuthorizeRouterData> for BluesnapPaymentsRequest {
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(item: &types::PaymentsAuthorizeRouterData) -> Result<Self, Self::Error> {
+        let auth_mode = match item.request.capture_method {
+            Some(enums::CaptureMethod::Manual) => BluesnapTxnType::AuthOnly,
+            _ => BluesnapTxnType::AuthCapture,
+        };
+        let payment_method = match item.request.payment_method_data.clone() {
+            api::PaymentMethod::Card(ccard) => Ok(PaymentMethodDetails::CreditCard(Card {
+                card_number: ccard.card_number.peek().clone(),
+                expiration_month: ccard.card_exp_month.peek().clone(),
+                expiration_year: ccard.card_exp_year.peek().clone(),
+                security_code: ccard.card_cvc.peek().clone(),
+            })),
+            _ => Err(errors::ConnectorError::NotImplemented(
+                "Unknown payment method".to_string(),
+            )),
+        }?;
+        Ok(Self {
+            amount: item.request.amount,
+            payment_method,
+            currency: item.request.currency,
+            soft_descriptor: item.description.clone(),
+            card_transaction_type: auth_mode,
+        })
+    }
+}
+
+#[derive(Debug, Serialize, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct BluesnapVoidRequest {
+    card_transaction_type: BluesnapTxnType,
+    transaction_id: String,
+}
+
+impl TryFrom<&types::PaymentsCancelRouterData> for BluesnapVoidRequest {
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(item: &types::PaymentsCancelRouterData) -> Result<Self, Self::Error> {
+        let card_transaction_type = BluesnapTxnType::AuthReversal;
+        let transaction_id = item.request.connector_transaction_id.to_string();
+        Ok(Self {
+            card_transaction_type,
+            transaction_id,
+        })
+    }
+}
+
+#[derive(Debug, Serialize, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct BluesnapCaptureRequest {
+    card_transaction_type: BluesnapTxnType,
+    transaction_id: String,
+    amount: Option<i64>,
+}
+
+impl TryFrom<&types::PaymentsCaptureRouterData> for BluesnapCaptureRequest {
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(item: &types::PaymentsCaptureRouterData) -> Result<Self, Self::Error> {
+        let card_transaction_type = BluesnapTxnType::Capture;
+        let transaction_id = item.request.connector_transaction_id.to_string();
+        Ok(Self {
+            card_transaction_type,
+            transaction_id,
+            amount: item.request.amount_to_capture,
+        })
+    }
+}
+
+// Auth Struct
+pub struct BluesnapAuthType {
+    pub(super) api_key: String,
+    pub(super) key1: String,
```

Comment:
key1 is not required for bluesnap as far as I am aware

Reply:
We are taking username and password from the merchant and then converting into base64 encoded api_key ourselves